database/store.py

```diff
@@ -26,45 +27,196 @@ def applications_index(self) -> dict:
                 "id": application.get("id"),
                 "account_id": application.get("account_id"),
                 "status": application.get("status"),
+                "account_id": application.get("account_id"),
                 "fund_id": application.get("fund_id"),
                 "round_id": application.get("round_id"),
+                "project_name": application.get("project_name", ""),
+                "last_edited": application.get("last_edited"),
+                "started_at": application.get("started_at"),
                 "date_submitted": application.get("date_submitted"),
-                "assessment_deadline": application.get("assessment_deadline"),
             }
             applications.update({application.get("id"): application_summary})
         return applications
 
-    def create_application(self, application):
-        fund_id = slugify(application["name"])
-        application_id, new_application = self.set_attributes(
-            fund_id, application
+    def create_application(self, account_id, fund_id, round_id):
+        sections = self._get_sections(fund_id, round_id)
+        application_id, new_application = self._set_attributes(
+            account_id, fund_id, round_id, sections
         )
         self._applications.update({application_id: new_application})
         return new_application
 
+    def _get_sections(self, fund_id: str, round_id: str):
+        fund = get_fund(fund_id)
+        fund_round = get_round(fund_id, round_id)
+        if fund and fund_round:
+            sections = fund_round_sections.get(":".join([fund_id, round_id]))
+            if not sections:
+                raise Exception(
+                    f"Could not find form sections for {fund_id} - {round_id}"
+                )
+            return sections
+        raise Exception(
+            f"Could not find fund round for {fund_id} - {round_id}"
+        )
+
+    def submit_application(self, application_id):
+        self._applications[application_id][
+            "date_submitted"
+        ] = datetime.datetime.now(datetime.timezone.utc).strftime(
+            "%Y-%m-%d %H:%M:%S"
+        )
+        self._update_statuses(application_id)
+        return self._applications[application_id]
+
+    def get_section(self, application_id, section_name):
+        return self._applications[application_id]["sections"][section_name]
+
+    def _find_value_by_key(self, data: dict, target):
+        for key, value in data.items():
+            if isinstance(value, dict):
+                yield from self._find_value_by_key(value, target)
```

Comment:
💯 ⛏️  This is really cool Harry, could we have some comments to explain your expert python skills? I really like the use of recursion and is instance!! <3

Reply:
🤔 Why have we used yield instead of return :)

Reply:
I have to agree - totally pointless

Reply:
Would be useful if we expected there to be multiple keys with matching names (which there could be in other dictionaries) but here we're only expecting there to be one